### PR TITLE
[Mobile Payments] Parse and rename eligibility flag

### LINKED ISSUE #4160: [Mobile Payments] Update WCPayAccount.init(from: Decoder)  to parse value for CPP eligibility flag
Update the code hat is parsing the WCPayAccount object to taken account the flag based by the server, that indicates if a store is eligible for card present payments. 

This depends on https://github.com/Automattic/woocommerce-payments-server/issues/725

If that PR to WCPay server didn't get merged on time, we could even hardcode a set of eligible siteIDs into this method.

### PR DESCRIPTION
**Pull request by ctarda**
Closes #4160 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️

There is ongoing work to pass down the eligibility flag from WCPay; https://github.com/Automattic/woocommerce-payments-server/pull/813

To be completely honest, I am not sure if submitting this now is actually a good thing, give we'd still need to come back to this in #4241 . My thinking was that it's better to get a bit closer to the final stage now and leave one line (in #4241) for later. But who knows.

The feature flag is hardcoded to true for now, but I've logged #4241 to revert that as soon as https://github.com/Automattic/woocommerce-payments-server/pull/813 is available

## Changes
* Parse the eligibility flag using the proper key, to integrate with the work done on the backend.
* Rename the eligibility flag to match the name it has on the backend.

I excluded unit tests, will add those with #4241 

## How to test
Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. WCPay 2.4.0

* Checkout the branch
* Connect to a reader
* Navigate to an eligible order. The "Collect Payment" button should be available in debug builds
* Navigate to Settings. The "Manage Card Readers" section should be available in debug builds.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Model/WCPayAccount.swift [hunk 1]
```diff
@@ -37,7 +38,7 @@ public struct WCPayAccount: Decodable {
         self.supportedCurrencies = supportedCurrencies
         self.country = country
         /// Hardcoded until support for this property is available in WCPay
-        self.canCollectPayments = true
+        self.isCardPresentEligible = true
```
Comment (ctarda) [inline Networking/Networking/Model/WCPayAccount.swift:41]: This is what we'd need to change in #4241 , and set the property to the value passed as parameter, instead f hardcoding it to true

**Comment by koke** [comment]
> should be available in debug builds

Is there something making the feature not available in release builds? Should we test for that?

**Comment by ctarda** [comment]
@koke yes, there is a feature flag: `cardPresentPayments`. The entry points to the feature in the UI ("Collect payment" button  and "Manage Card Readers" section in Settings) should be gated with that feature flag as well

**Review by koke — APPROVED** [review]
🤦🏽 I was looking at the line of code where it's checked and somehow missed it. Tested the change and the feature is still available

**Pull request merged**